src/poetry/utils/env/python/manager.py

```diff
@@ -252,10 +244,8 @@ def get_system_python(cls) -> Python:
 
     @classmethod
     def get_by_name(cls, python_name: str) -> Python | None:
-        if Path(python_name).exists():
-            with contextlib.suppress(ValueError):
-                # if it is a path try assuming it is an executable
-                return cls.from_executable(python_name)
+        if python := ShutilWhichPythonProvider.find_python_by_name(python_name):
+            return cls(python=python)
```

Comment:
The value error suppression should remain I think. This is to avoid accidentally using a python that does not work (broken installations etc).